### ISSUE TITLE
Algorithm 1 in paper

### ISSUE THREAD

**Issue by bkj** [HIPS/hypergrad#3]
Hi All --

In the paper [1], Algorithm 1 is

<img width="344" alt="screen shot 2017-09-07 at 9 33 51 am" src="https://user-images.githubusercontent.com/6086781/30165936-cda83c62-93af-11e7-9762-e6283eaa5e11.png">

Should line 6 actually say
```
w[t+1] = w[t] + alpha[t] * v[t + 1]
```

Otherwise, you're updating the weights w/ a velocity that doesn't consider the most recent gradient.  AFAICT, the `hypergrad` code does the latter, but wanted to see if I could verify whether this was indeed a typo.

(Reason for this is I'm trying to derive Algorithm 2, so was looking very closely at the notation.  If there's any insight into Algorithm 2 that someone could provide, that'd be much appreciated.  Specifically lines 9-13, where the gradients w.r.t. the hyperparameters are calculated.)

[1] https://arxiv.org/pdf/1502.03492.pdf

**Comment by duvenaud** [comment]
Yes, this is a typo - the hypergrad code has the correct implementation, as you noticed.  Thanks for pointing it out!  Algorithm 2 also has a couple of typos in the paper but not in the code.

**Comment by bkj** [comment]
Great -- thanks!  Will use the code as a reference, in that case.

**Comment by bkj** [comment]
A related question -- the version of SGD w/ momentum implemented in `hypergrad` is different from the standard implementation.  `hypergrad` has:
```
velocity = momentum * velocity - (1.0 - momentum) * g
x = x + learning_rate * velocity
```
vs other place (eg the NAG paper, CS231 from Stanford) have
```
velocity = momentum * velocity - learning_rate * g
x = x + velocity
```

@duvenaud Do you have any insight on why the `hypergrad` implementation differs?

~ Ben

**Comment by mattjj** [comment]
Correct me if I'm wrong but I think your latter question was answered in [autograd#289](https://github.com/HIPS/autograd/issues/289). Maybe this issue can be closed?

**Comment by bkj** [comment]
It was -- thanks.  Apologies for the double post.
